functional-tests/fn_bridge_deposit_reclaim_drt_seen.py

```diff
@@ -0,0 +1,171 @@
+        refund_btc_balance = get_balance(
+            refund_addr,
+            btc_url,
+            btc_user,
+            btc_password,
+        )
+        self.logger.debug(f"Refund BTC balance (before takeback): {refund_btc_balance}")
+        assert refund_btc_balance == initial_refund_btc_balance, "BTC balance has changed"
+
+        # Spend the take back path
+        take_back_tx = bytes(
+            take_back_transaction(refund_addr, bridge_pk, btc_url, btc_user, btc_password)
+        ).hex()
+        self.logger.debug("Take back tx generated")
+
+        # Send the transaction to the Bitcoin network
+        txid = btcrpc.proxy.sendrawtransaction(take_back_tx)
+        self.logger.debug(f"sent take back tx with txid = {txid} for address {el_address}")
+        btcrpc.proxy.generatetoaddress(2, UNSPENDABLE_ADDRESS)
+        time.sleep(1)
+
+        # Make sure that the BTC recovery address has 0 BTC
+        btc_balance = get_balance_recovery(
+            recovery_addr,
+            bridge_pk,
+            btc_url,
+            btc_user,
+            btc_password,
+        )
+        self.logger.debug(f"DRT BTC balance (after takeback): {btc_balance}")
+        assert btc_balance == 0, "BTC balance is not zero"
+
+        # Make sure that the BTC refund address has the expected balance
+        refund_btc_balance = get_balance(
+            refund_addr,
+            btc_url,
+            btc_user,
+            btc_password,
+        )
+        self.logger.debug(f"User BTC balance (after takeback): {refund_btc_balance}")
+        expected_balance = 5 * DEPOSIT_AMOUNT - TAKE_BACK_FEE
+        assert refund_btc_balance >= expected_balance, "BTC balance is not as expected"
+
+        # Get the balance of the EL address after the deposits
+        balance_post = int(rethrpc.eth_getBalance(el_address), 16)
+        self.logger.debug(f"Strata Balance after deposits: {balance_post}")
+        assert balance_post == 0, "Strata balance is not zero"
+
+        return True
+
+    def make_drt(self, ctx: flexitest.RunContext, el_address, musig_bridge_pk):
+        """
+        Deposit Request Transaction
+        """
+        # Get relevant data
+        btc = ctx.get_service("bitcoin")
+        btcrpc: BitcoindClient = btc.create_rpc()
+        btc_url = btcrpc.base_url
+        btc_user = btc.props["rpc_user"]
+        btc_password = btc.props["rpc_password"]
+
+        # Create the deposit request transaction
+        tx = bytes(
+            deposit_request_transaction(
+                el_address, musig_bridge_pk, btc_url, btc_user, btc_password
+            )
+        ).hex()
+        self.logger.debug(f"Deposit request tx: {tx}")
+
+        # Send the transaction to the Bitcoin network
+        txid = btcrpc.proxy.sendrawtransaction(tx)
+        self.logger.debug(f"sent deposit request with txid = {txid} for address {el_address}")
+        time.sleep(1)  # I don't know how to get rid of this sleep
```

Comment:
That would be ideal.
But this transaction is not made "in-wallet" with respect to `bitcoind`.
It is made in the `python-strata-utils`.
Hence I cannot use [`gettransaction`](https://developer.bitcoin.org/reference/rpc/gettransaction.html) and [`getrawtransaction`](https://developer.bitcoin.org/reference/rpc/getrawtransaction.html) only works with mempool transactions, this is already mined.

That's why I don't know how to get rid of the `time.sleep`.

Do you have a good suggestion?

Reply:
Does it not work when you set `-txindex=1` when you instantiate the test bitcoind instance? Also, isn't DRT signed with a wallet available on the bitcoind instance? Easiest would be to use the same bitcoind instance but separate wallets so that we can do this. Another option would be to query the current best block height after publishing DRT  and then wait for some number of blocks to be published. 

But if none of these are feasible, leaving a comment regarding what we're waiting for here would be okay, I guess.

Reply:
I got rid of all `time.sleep` in f84f072913c902683ae2caf644eb3b37f010b014
🎉